src/components/DocumentUtility/DocumentUtility.tsx

```diff
@@ -14,21 +16,42 @@ interface DocumentUtilityProps {
 
 export const DocumentUtilityUnStyled = ({ document, handleSharingToggle, className }: DocumentUtilityProps) => {
   const fileName = getData(document).name;
+  const qrcodeUrl = getData(document)?.links?.self?.href ?? "";
+
+  const qrCodePopover = (url: string) => (
+    <Popover id="qr-code-popover" style={{ borderRadius: 0, border: "1px solid #DDDDDD" }}>
+      <Popover.Content style={{ padding: 0 }}>
+        <QRCode bgColor="#FFFFFF" fgColor="#000000" level="Q" style={{ width: 200, padding: "10px" }} value={url} />
+      </Popover.Content>
+    </Popover>
+  );
 
   return (
     <div className={`${className}`}>
       <div className="container-custom">
         <div className="row no-gutters">
           <div className="col-auto ml-auto">
-            <ButtonIconWhiteBlue onClick={() => window.print()}>
+            <OverlayTrigger trigger="click" placement="bottom-end" overlay={qrCodePopover(qrcodeUrl)}>
+              <ButtonIconWhiteBlue aria-label="document-utility-qr-button" hidden={!qrcodeUrl}>
+                <SvgIcon strokeWidth="0.5" fill="currentColor">
+                  <SvgIconQRCode />
+                </SvgIcon>
+              </ButtonIconWhiteBlue>
+            </OverlayTrigger>
```

Comment:
```suggestion
            {qrcodeUrl && (
              <OverlayTrigger trigger="click" placement="bottom-end" overlay={qrCodePopover(qrcodeUrl)}>
                <ButtonIconWhiteBlue aria-label="document-utility-qr-button">
                  <SvgIcon strokeWidth="0.5" fill="currentColor">
                    <SvgIconQRCode />
                  </SvgIcon>
                </ButtonIconWhiteBlue>
              </OverlayTrigger>
            )}
```

how about this? then maybe no need hidden props for button too already

Reply:
fixed :smile: